src/ifp/src/InitFloorplan.cc

```diff
@@ -569,4 +579,80 @@ void InitFloorplan::makeTracks(odb::dbTechLayer* layer,
   grid->addGridPatternY(origin_y, y_track_count, y_pitch);
 }
 
+void InitFloorplan::makeTracksNonUniform(odb::dbTechLayer* layer,
+                                         int x_offset,
+                                         int x_pitch,
+                                         int y_offset,
+                                         int y_pitch,
+                                         int first_last_pitch)
+{
+  if (layer->getDirection() != dbTechLayerDir::HORIZONTAL) {
+    logger_->error(IFP, 40, "Non horizontal layer uses property LEF58_PITCH.");
+  }
+  auto rows = block_->getRows();
+  int cell_row_height = rows.begin()->getSite()->getHeight();
```

Comment:
You cannot assume the first row will not be of type PAD, you need to find the appropriate row